### PR TITLE
Send workspace_config only if changed

### PR DESCRIPTION
**Pull request by hrsh7th**
Fixes https://github.com/prabirshrestha/vim-lsp/issues/1107

### CODE REVIEW

File: autoload/lsp.vim [hunk 1]
```diff
@@ -616,12 +616,12 @@ endfunction
 function! s:ensure_conf(buf, server_name, cb) abort
     let l:server = s:servers[a:server_name]
     let l:server_info = l:server['server_info']
-    if has_key(l:server_info, 'workspace_config')
-        let l:workspace_config = l:server_info['workspace_config']
+    if has_key(l:server_info, 'workspace_config') && !get(l:server_info, '_workspace_config_sent', v:false)
```
Comment (rhysd) [inline autoload/lsp.vim:619]: This means didChangeConfiguration is sent only once and never sent again. `lsp#update_workspace_config` no longer works. Is it ok?

Comment (hrsh7th) [reply autoload/lsp.vim:619]: I think it should be work as expected because `lsp#update_workspace_config` will re-set flags.

Comment (rhysd) [reply autoload/lsp.vim:619]: Hm, I didn't see how they are reset.

The implementation of `lsp#update_workspace_config` is as follows:

```vim
function! lsp#update_workspace_config(server_name, workspace_config) abort
    let l:server = s:servers[a:server_name]
    let l:server_info = l:server['server_info']
    if has_key(l:server_info, 'workspace_config')
        call s:merge_dict(l:server_info['workspace_config'], a:workspace_config)
    else
        let l:server_info['workspace_config'] = a:workspace_config
    endif
    call s:ensure_conf(bufnr('%'), a:server_name, function('s:Noop'))
endfunction
```

It merges new config into existing one and immediately calls `s:ensure_conf`. I couldn't find where the `_workspace_config_sent` flag is reset. Would you explain how the flag is reset?

Comment (rhysd) [reply autoload/lsp.vim:619]: Ah, I overlooked https://github.com/prabirshrestha/vim-lsp/pull/1137/commits/c473b5c8cfdbc8cedcdb6c4ceaa746db61a74dca#diff-dd9e05ae74405827f8c7e93a94069e8386722ce20e643642aa37c4049f24b070R1198. Thanks for the explanation

**Review by rhysd — APPROVED** [review]

**Comment by hrsh7th** [comment]
If someone reports any problem on this, I will tackle it.

**Pull request merged**